src/reyna/Blackout/BlackoutTime.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Reyna
+{
+    using System;
+    using System.Globalization;
+
+    public class BlackoutTime
+    {
+        public bool CanSendAtTime(DateTime now, string ranges)
+        {
+            if (String.IsNullOrEmpty(ranges))
+            {
+                return true;
+            }
+
+            string[] rangesSplit = ranges.Split(',');
+            foreach (string range in rangesSplit)
+            {
+                TimeRange timeRange = this.ParseTime(range);
+                Time timeNow = new Time(now.Hour, now.Minute);
+                if (timeRange.Contains(timeNow))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public TimeRange ParseTime(string range)
+        {
+            string[] rangeSplit = range.Split('-');
+            string format = "HH:mm";
+
+            DateTime from = DateTime.ParseExact(rangeSplit[0], format, CultureInfo.InvariantCulture);
+            DateTime to = DateTime.ParseExact(rangeSplit[1], format, CultureInfo.InvariantCulture);
```

Comment:
We need to check that the resulted array after split is minimum length of 2, otherwise when parse something like 1200 we will get exception.
@Youhana-Hana